src/main/java/com/google/firebase/internal/RetryConfig.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.internal;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.client.util.BackOff;
+import com.google.api.client.util.ExponentialBackOff;
+import com.google.api.client.util.Sleeper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Configures when and how HTTP requests should be retried.
+ */
+public final class RetryConfig {
+
+  private static final int INITIAL_INTERVAL_MILLIS = 500;
+
+  private final List<Integer> retryStatusCodes;
+  private final boolean retryOnIOExceptions;
+  private final int maxRetries;
+  private final Sleeper sleeper;
+  private final ExponentialBackOff.Builder backOffBuilder;
+
+  private RetryConfig(Builder builder) {
```

Comment:
Feel free to ignore, but I think you could use `@AutoValue` here since this is internal and doesn't leak to our customers. You are already getting it for free as part of the Firestore dependencies.

Reply:
We don't use it anywhere in the codebase at the moment. I'd like to not introduce another dependency at this point.